### PR TITLE
RE-328: Pull in upstream

### PR DESCRIPTION
**Pull request by scthi**
This PR brings in the upstream changes from https://github.com/trigger-corp/trigger.io-tabs into our sb_tabs.

So it's basically the upstream version (we are having atm in our app) plus the upload support added by @hassaanz in MA-8080

The corresponding frontend PR: https://github.com/Staffbase/frontend/pull/401 (draft atm)

### CODE REVIEW

File: inspector/an-inspector/ForgeModule/src/io/trigger/forge/android/modules/tabs/API.java [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
-			modalViews.get(modal).addButton(task, params);
-		}
-	}
-
-	public static void removeButtons(final ForgeTask task, @ForgeParam("modal") final String modal) {
-		if (modalViews.get(modal) != null) {
-			modalViews.get(modal).removeButtons(task);
-		}
-	}
-}
```
Comment (haed) [inline inspector/an-inspector/ForgeModule/src/io/trigger/forge/android/modules/tabs/API.java:43]: I can see no difference here. On creating PRs to public or external repos unnecessary code structure changes should be avoided to keep merge conflicts and review overhead at a minimum.

File: inspector/an-inspector/ForgeModule/src/io/trigger/forge/android/modules/tabs/ModalView.java [hunk 2]
```diff
@@ -252,6 +254,9 @@ public void run() {
                 view.addView(webView, new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT, 1));
                 webView.loadUrl(url);
 
+                // update content insets
```
Comment (haed) [inline inspector/an-inspector/ForgeModule/src/io/trigger/forge/android/modules/tabs/ModalView.java]: This comment makes no sense, it's the same text like the line below.

**Review by haed — CHANGES_REQUESTED** [review]

File: module/tests/interactive.js [hunk 3]
```diff
@@ -2,8 +2,26 @@
...
+        });
+    }, function (e) {
+        ok(false, "API call failure: "+e.message);
+        start();
+    });
+});
+
 asyncTest("Open tab with downloads", 1, function() {
-    forge.tabs.open("https://req-playground-ahghagcnmb.now.sh/downloads", function () {
+    forge.tabs.open("https://req-playground.scthi.tech/downloads", function () {
```
Comment (Drako) [inline module/tests/interactive.js:24]: nice domain :)

File: inspector/an-inspector/ForgeModule/src/io/trigger/forge/android/modules/tabs/ModalView.java (hunk 2 above)
Comment (hassaanz) [reply inspector/an-inspector/ForgeModule/src/io/trigger/forge/android/modules/tabs/ModalView.java]: I removed this in 489868f7ec198c76464d08526daa5f7d5efe9301

File: inspector/an-inspector/ForgeModule/src/io/trigger/forge/android/modules/tabs/API.java (hunk 1 above)
Comment (hassaanz) [reply inspector/an-inspector/ForgeModule/src/io/trigger/forge/android/modules/tabs/API.java:43]: There is something wrong with line endings here not related to auto formatting. I am currently investigating it.

Comment (hassaanz) [reply inspector/an-inspector/ForgeModule/src/io/trigger/forge/android/modules/tabs/API.java:43]: I looked into it and the line endings don't affect the "annotate" feature. I would resolve this for now and might change if Trigger Guys complain about it.

**Review by hassaanz — APPROVED** [review]

**Review by haed — APPROVED** [review]

**Review by ugeshmalempat — APPROVED** [review]

**Pull request merged**